### DISCUSSION TITLE
bleak data reading rate through notification  with one uuid (57 samples per second ) and with 2 uuid subscribe (72 samples per second which is 36 sample of each uuid)

### DISCUSSION THREAD

**Discussion by siki411** [hbldh/bleak#1593] [Q&A]
* bleak version: 0.22.2
* Python version: 3.8.10
* Operating System: windows 10
* BlueZ version (`bluetoothctl -v`) in case of Linux:

### Description
so i am new to python and bleak so i have implemented notification reading with simplest possible way where i am just scanning the device and then start notify function with call back in order to read data  but the problem is that i am not been able to read data fastly through notification.. with single uuid notification i am able to read 57 samples per second and if i go with 2 uuid notification and if i subscribe two notification then i been able to read 72 samples per second which means almost 36 samples i am getting from each uuid (32*2=72) but i have to read almost 100 samples per second now i dont know whether its bleak issue or hardware but with the same hardware but with uart i was able to read 100 samples per second. i would be really appreciate if you can guide me or recommand something
 thanks

**Comment by dlech** [comment]
I would recommend logging Bluetooth packets using a BLE sniffer to see what is actually going over the air. This lets you see things like the connection interval that can affect throughput. 

Most likely, this is a low-level hardware configuration issue so can't be fixed by anything in Bleak.
